network/peer-monitoring-service/server/src/lib.rs

```diff
@@ -204,6 +251,30 @@ impl Handler {
         ))
     }
 
+    fn get_node_information(&self) -> Result<PeerMonitoringServiceResponse, Error> {
+        // Get the node information
+        let git_hash = aptos_build_info::get_git_hash();
+        let current_time: Instant = self.time_service.now();
+        let uptime = current_time.duration_since(self.start_time);
+        let (highest_synced_epoch, highest_synced_version) =
+            self.storage.get_highest_synced_epoch_and_version()?;
+        let ledger_timestamp_usecs = self.storage.get_ledger_timestamp_usecs()?;
+        let lowest_available_version = self.storage.get_lowest_available_version()?;
+
+        // Create and return the response
+        let node_information_response = NodeInformationResponse {
+            git_hash,
```

Comment:
stupid question, why do we want nodes to know git_hashes of other nodes? :)

is it just for metrics, or are we going to prefer peers with same git_hash , or?

Reply:
Yeah, it's mostly for observability/debugging purposes (e.g., we're running a node and we're seeing bad peer behaviour -- it would be good to know if the `git_hash` matches what we expect, etc.). Similarly, we could use it in some extreme scenarios to handle network upgrades (i.e., prioritize upgraded nodes over non-upgraded ones, but that'd be rarer, I think 😄)